experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -392,11 +396,29 @@ export class FieldProxyTarget extends ProxyTarget<FieldAnchor> implements Editab
 		);
 	}
 
+	public treeStatus(): TreeStatus {
+		if (this.isFreed()) {
+			return TreeStatus.Deleted;
+		}
+		const fieldAnchor = this.getAnchor();
+		const parentAnchor = fieldAnchor.parent;
+		// If the parentAnchor is undefined it is a detached field.
+		if (parentAnchor === undefined) {
+			return keyAsDetachedField(fieldAnchor.fieldKey) === rootField
+				? TreeStatus.InDocument
+				: TreeStatus.Removed;
+		}
+		const parentAnchorNode = this.context.forest.anchors.locate(parentAnchor);
+		// As the "parentAnchor === undefined" case is handled above, parentAnchorNode should exist.
+		assert(parentAnchorNode !== undefined, "parentAnchorNode must exist.");
+		return treeStatusFromPath(parentAnchorNode);
```

Comment:
I changed the logic using anchors instead of cursors, but still couldn't get around using an assert statement as the `locate()` method can still return undefined. Since we already checked that `parentAnchor` is not undefined (from the if statement above), would it be safe to keep this assert here?

Reply:
You shouldn't be doing `const parentAnchor = fieldAnchor.parent;` nor asserting `fieldAnchor.parent !== undefined`.

That case is valid, and you should handle it with an if statement, not crash when its the case. Your goal here is to find the key of the root field. When your path does not have a parent, that means you are at the root, and need to check that field key. By throwing away that field key by only using the parent of the fieldPath from the anchor, you make this impossible.

Reply:
This is fine: you accurately describe why the assert should always pass (and I agree its a valid assumption). The fact that there isn't a good way to avoid this assert is a hint that maybe we could provide better APIs since it means we have assumptions that arn't captured in the types. For example if we had:

```typescript
export function locateFieldAnchor(locator: AnchorLocator, anchor: FieldAnchor): FieldUpPath<AnchorNode> { ... }
```

You could call locateFieldAnchor at the beginning, then only check for undefined parent in once place (in the returned FieldUpPath) instead of in both the if and assert.

If you write that function (after https://github.com/microsoft/FluidFramework/pull/16950 goes in), it will also make testing a little easier since slightly more logic is moved into a simpler to test place.

That said, this code is fine as is, so feel free to leave if you want.

Reply:
Updated assert to if statement due to the following reason:

The anchors.locate method seems to return undefined when its corresponding PathNode does not have the status Status.Alive. I updated the assert statement to use an if statement instead, and return TreeStatus.Delete in this case, but I am not entirely sure if this is correct.

Based on the documentation there seems to be 2 other statuses Status.Disposed and Status.Dangling. Status.Dangling seems to correspond to a deleted node. But does Status.Disposed mean that it's removed (but invalid)? If that's the case, we may need to distinguish between the two cases, but there seems to be no good way to get the PathNode status.

Should an additional method along the lines of getPathNodeStatus(anchor:Anchor) be created in AnchorSet so that we could make this distinction?

Reply:
You should never see any status other than alive since if you have an ProxyTarget that has not been freed: the ProxyTarget holds onto an anchor, which ensures the AnchorNode doesn't get disposed, and if the AnchorNode does become dangling, it will have triggered its "afterDelete" event, which frees the ProxyTarget.